ItemPool.py

```diff
@@ -1352,6 +1364,13 @@ def get_pool_core(world):
             pool.remove(junk_item)
             pool.append(pending_item)
 
+    # Add non-repeatable items to pool if no repeatable merchant or cow locations are randomized
+    if not world.shuffle_medigoron_carpet_salesman: #and world.shopsanity in ('off', '0') and world.shuffle_scrubs == 'off' and not world.shuffle_cows:
+        for item in pool:
```

Comment:
This needs to iterate over a copy, or (preferable:) iterate with indices (e.g. enumerate) and re-assign directly to `pool[i]`. Removing and appending are bad ideas while iterating.

Reply:
I'm going to remove this check entirely because it doesn't really do anything. If any non-optional checks were made repeatable (such as letting you get the repeatable item again from redoing minigames instead of a purple rupee) then it would have to be removed anyway.